docs/code/component-architecture/README.md

```diff
@@ -1,9 +1,132 @@
 # Component Architecture
 
-TODO: Component Directory Structure
-TODO: Naming Components and their Props
-TODO: Prop Docs
-TODO: Class v Functional Components
```

Comment:
Would it be worth adding something like "By default, we try to use [Functional Components](https://reactjs.org/docs/components-and-props.html#function-and-class-components) unless a Class Component is required." possibly to `component.js`

Reply:
I've actually put together a detailed reference on how to structure components in the syntax styleguide page. I added one point there and put a reference in the `component.js` section here.